libs/sdk-core/src/breez_services.rs

```diff
@@ -278,36 +277,32 @@ impl BreezServices {
     ///
     /// # Arguments
     ///
-    /// * `node_id` - The destination node_id
-    /// * `amount_sats` - The amount to pay in satoshis
+    /// * `req` - Request parameters for sending a spontaneous payment
     pub async fn send_spontaneous_payment(
         &self,
-        node_id: String,
-        amount_sats: u64,
-    ) -> SdkResult<Payment> {
+        req: SendSpontaneousPaymentRequest,
+    ) -> SdkResult<SendPaymentResponse> {
         self.start_node().await?;
         let payment_res = self
             .node_api
-            .send_spontaneous_payment(node_id.clone(), amount_sats)
+            .send_spontaneous_payment(req.node_id.clone(), req.amount_msat)
             .await;
-        self.on_payment_completed(node_id, None, payment_res).await
+        let payment = self
+            .on_payment_completed(req.node_id, None, payment_res)
+            .await?;
+        Ok(SendPaymentResponse { payment })
     }
 
     /// Second step of LNURL-pay. The first step is `parse()`, which also validates the LNURL destination
     /// and generates the `LnUrlPayRequestData` payload needed here.
     ///
-    /// This call will validate the given `user_amount_sat` and `comment` against the parameters
+    /// This call will validate the given `amount_msat` and `comment` against the parameters
     /// of the LNURL endpoint (`req_data`). If they match the endpoint requirements, the LNURL payment
     /// is made.
     ///
     /// This method will return an [anyhow::Error] when any validation check fails.
```

Comment:
The documentation has to reference LnUrlPayRequest imo.

Reply:
Done